packages/react-components/react-avatar/src/utils/getPartitionedAvatarGroupItems.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Get the inline items and overflowing items based on the array of AvatarGroupItems needed for AvatarGroup.
+ *
+ * @param items - Array of AvatarGroupItems
+ * @param maxItems - Maximum number of items to show inline
+ *
+ * @returns Two arrays of AvatarGroupItems splitted between the inline items and the overflowing items
+ * based on maxItems.
+ *
+ * NOTE: Each item in the array must be a single AvatarGroupItem. If an item in the array contains a fragment with
+ * two avatars, the items will not be partitioned correctly.
+ */
+export const getPartitionedAvatarGroupItems = (items: JSX.Element[], maxItems = 5) => {
+  // When there are more items than maxItems, we need to subtract one to make space for the overflow indicator.
+  const numOfInlineItems = items.length > maxItems ? maxItems - 1 : maxItems;
+
+  const inlineItems = items.slice(-numOfInlineItems);
+  const overflowItems = items.slice(0, -numOfInlineItems);
+
+  return { inlineItems, overflowItems };
```

Comment:
Are you sure this `overflowItems` collection will always be accurate ? if you are using `react-overflow` then the actual number of overflowing items will be decided based on the available space

Reply:
We can probably fix it in a follow up PR, but I agree with @ling1726 as it's unsafe to make such assumptions:

![image](https://user-images.githubusercontent.com/14183168/182124801-0cc215a8-a683-421f-b231-220d0d66fe34.png)

Reply:
AvatarGroup doesn't use react-overflow, therefore this assumption would be correct. Here's a [test](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhjAvDA2gBgDQEYC6BuAKFElgCMlVMsMAmfI8aGYC9bWjAZgwBYMBWDADYMAdnoMSMAA5wATlACWS8AEEAbnCjyA4nJABXaQEkoAUwC2ECgAplliAC4Ycs3AAm4ADYBPeGB8UHAwYLzgfQygAfmcAcmlFM1iYAB8YWOg4YABrZLSM6VcPWJDFMC8ys1MHAGFDMFhkfgBKJAA+GABvAhhmRlhFCAAFRIowiINGxGR4xNjCXuImCzgAD2NyyuqrCkGRsxgomE4YZzKKsCrzKzqDBoW+qXPK24aKAFobFfXNy+3rd4wOzXCAAOi8ZjAAHMoAALGAdb4bC5XByHGBYU4wNDNZqEHqhMwDX6onbIIIPJawEDqMxyABmXhAAHd-rthqMjvYdmcQaCIBVgGYbJgYM9Lq8oHiHop6UC9olWt1er1xaTrMhuWCBYohSKQkiSf88QSAL4wMxeCAHZUqtVszV8nV6tWSk29U0EAmuKAGORgLpio0gkI0umMlls02EaNepYgCHgkBQmyyBTKRRqTTaOR6QwmEE2OC4wjxxNMlNppQqMAaLS6fRGf42UglyQQBNmJOV+TVzO17MN-PN4Btstdis2WZmOIhKsZrP13ONgsOIshaexMeMTvdqcJGfpOe9hcDpd5puF0gbg9bk3jvfT2cyE81us5i+rqw2YA3uYloA) of it, the main idea of this function is to just split the data/elements with the correct behavior expected from an AvatarGroup. This splitting does not rely on available space, but on how many avatars should be displayed inline.

Reply:
@sopranopillow  I wonder if we should rename AvatarGroupOverflow (e.g. to AvatarGroupPopover), to avoid confusion with the react-overflow package, since it's not related?
[Edit] Don't let this block this PR. It can be addressed in a future PR if we decide to go ahead with the rename.